R/plot.data_table_threads_benchmark.R

```diff
@@ -16,46 +16,56 @@
-#' benchmarkData <- findOptimalThreadCount(1000, 10)
+#' benchmarkData <- data.table.threads::findOptimalThreadCount(1000, 10)
 #' # Generating speedup plots based on the data collected above:
 #' plot(benchmarkData)
 #' }
 
 plot.data_table_threads_benchmark <- function(x, ...)
 {
-  df <- x
-  rownames(df) <- NULL
-  df$speedup <- df$medianTime[df$threadCount == 1] / df$medianTime
-
-  setDT(df)
-  maxSpeedup <- df[, .(threadCount = threadCount[which.max(speedup)], speedup = max(speedup)), by = expr]
-  idealSpeedup <- seq(1, getDTthreads())
-  idealSpeedupData <- data.frame(threadCount = 1:getDTthreads(), speedup = idealSpeedup)
-  subOptimalSpeedupData <- data.frame(threadCount = seq(1, getDTthreads(), length.out = getDTthreads()), speedup = seq(1, getDTthreads()/2, length.out = getDTthreads()))
-
-  closestPoints <- data.frame()
-  for(i in unique(df$expr)) {
-    dfSubset <- df[df$expr == i, ]
-    suboptimalSubset <- subOptimalSpeedupData[subOptimalSpeedupData$threadCount %in% dfSubset$threadCount, ]
-    closestPoint <- dfSubset[which.max(dfSubset$speedup - suboptimalSubset$speedup), ]
-    closestPoints <- rbind(closestPoints, closestPoint)
+  x[, `:=`(speedup = medianTime[threadCount == 1] / medianTime, type = "Measured"), by = expr]
+
+  systemThreadCount <- getDTthreads()
+  functions <- unique(x$expr)
+
+  speedupData <- data.table(
+    expr = rep(functions, each = systemThreadCount),
+    threadCount = rep(c(1:systemThreadCount, seq(1, systemThreadCount, length.out = systemThreadCount)), length(functions)),
+    speedup = c(rep(seq(1, systemThreadCount), length(functions)), rep(seq(1, systemThreadCount / 2, length.out = systemThreadCount), length(functions))),
+    type = rep(c("Ideal", "Recommended"), each = systemThreadCount * length(functions))
+  )
+
+  for(col in setdiff(names(x), names(speedupData))) 
+  {
+    speedupData[, (col) := NA]
   }
 
-  ggplot(df, aes(x = threadCount, y = speedup)) +
-    geom_line(aes(linetype = "Measured")) +
-    geom_line(data = idealSpeedupData, aes(x = threadCount, y = speedup, linetype = "Ideal"), color = "black") +
-    geom_line(data = subOptimalSpeedupData, aes(x = threadCount, y = speedup, linetype = "Sub-optimal"), color = "black") +
-    geom_point(data = closestPoints, aes(x = threadCount, y = speedup, shape = "Recommended"), color = "black", size = 2) +
-    geom_point(data = maxSpeedup, aes(x = threadCount, y = speedup, shape = "Best performing"), color = "red", size = 2) +
-    geom_text(data = closestPoints, aes(label = threadCount), vjust = -0.5, size = 4, na.rm = TRUE) +
-    geom_text(data = maxSpeedup, aes(label = threadCount), vjust = -0.5, size = 4, na.rm = TRUE) +
-    geom_ribbon(aes(ymin = speedup - 0.3, ymax = speedup + 0.3), alpha = 0.5) +
+  maxSpeedup <- x[, .(threadCount = threadCount[which.max(speedup)], speedup = max(speedup), type = "Ideal"), by = expr]
+
+  combinedLineData <- rbind(speedupData, x, use.names = TRUE, fill = TRUE)
```

Comment:
not sure why you need to use.names and fill? Please add a comment to explain what columns are missing in which variable.